### PR TITLE
refactor package structure

### PR DESCRIPTION
**Pull request by lgarithm**
- splitted `rchannel` into `rchannel/connection`, `rchannel/server`, `rchannel/client`, `rchannel/handler`
- splitted `kungfu` into `kungfu/peer`, `kungfu/session`
- renamed `kungfubase`, `kungfuconfig`, `kungfurun` to `kungfu/base`, `kungfu/config`, `kungfu/runner`.
- moved `rchannel/router.go` into `kungfu/peer/router.go`

### CODE REVIEW

**Comment by lgarithm** [comment]
@luomai ready to merge

**Comment by luomai** [comment]
awesome work! Let me have a look.

File: tests/go/cmd/kungfu-test-public-apis/kungfu-test-public-apis.go [hunk 1]
```diff
@@ -5,33 +5,34 @@ import (
...
 		fmt.Printf("# test: %d\n", i)
-		t(kungfu)
+		t(p)
 	}
 }
 
-func testGetPeerLatencies(kungfu *kf.Kungfu) {
-	sess := kungfu.CurrentSession()
+func testGetPeerLatencies(peer *peer.Peer) {
+	sess := peer.CurrentSession()
```
Comment (luomai) [inline tests/go/cmd/kungfu-test-public-apis/kungfu-test-public-apis.go]: What is the difference between peer.CurrentSession and defaultPeer.CurrentSession?

Comment (lgarithm) [reply tests/go/cmd/kungfu-test-public-apis/kungfu-test-public-apis.go]: `defaultPeer` is only used in C API.

**Pull request merged**